pkg/reconcilermanager/controllers/reposync_controller_test.go

```diff
@@ -281,6 +281,7 @@ func setupNSReconciler(t *testing.T, objs ...client.Object) (*syncerFake.Client,
 		fakeClient,
 		controllerruntime.Log.WithName("controllers").WithName("RepoSync"),
 		s,
+		false,
```

Comment:
Can you add unit test when allowVerticalScale is set to true?

Reply:
unit test has been added. The unit test will 
1. Add allowVerticalScale to test reconciler to mock real allowVerticalScale setting
2. Update the resource and replica directly in the fakeClient
3. Reconcile
4. Validate that resource update are maintained, but replica update is reverted